src/Functions/DateTimeTransforms.h

```diff
@@ -185,6 +193,14 @@ struct ToStartOfMonthImpl
     {
         return time_zone.toFirstDayNumOfMonth(DayNum(d));
     }
+    static inline Int64 execute_compat(Int64 t, const DateLUTImpl & time_zone)
```

Comment:
Maybe I miss something but should not the existing implementation (l. 180ff) be labeled `compat` whereas the new implementation (here) is the future?

Reply:
@rschu1ze and @ilejn 
My current variant is to simply replace:
is_compat by is_extended_result

and 
execute_compat() by execute_extended()  (or execute_extended_result())

It will keep code changes minimal. 
What do you think?

Reply:
OK. Thank you for clarification. I will use extended_result for 3 of them.

Reply:
I currently use:
enable-extended-results-for-datetime-functions option
bool is_extended_result
execute_extended_result()